python/ray/experimental/ui.py

```diff
@@ -275,79 +277,162 @@ def task_timeline():
-        assert os.path.exists(trace2html)
-        return trace2html
+                   "-C",
+                   catapult_home,
+                   "rev-parse",
+                   "--is-inside-work-tree"]
+            subprocess.check_call(cmd)
+
+            return catapult_home
+        except subprocess.CalledProcessError:
+            # Error on non-zero exit code (e.g. - ".git not found")
+            pass
+
+        print("Cloning catapult to {}.".format(catapult_home))
+        cmd = ["git",
+               "clone",
+               "https://github.com/catapult-project/catapult.git",
+               catapult_home]
+        subprocess.check_call(cmd)
+
+        # Checks out the commit associated with allowing different arrow
+        # colors. This can and should be removed after catapult's next
+        # release.
+        print("Checking out commit {}.".format(catapult_commit))
+        cmd = ["git", "-C", catapult_home, "checkout", catapult_commit]
+        subprocess.check_call(cmd)
+
+        return catapult_home
 
     def handle_submit(sender):
         tmp = tempfile.mktemp() + ".json"
         tmp2 = tempfile.mktemp() + ".html"
+        html_file = tmp2
 
+        # Determine whether task components should be displayed or not.
         if breakdown_opt.value == breakdown_basic:
             breakdown = False
         elif breakdown_opt.value == breakdown_task:
             breakdown = True
         else:
-            raise ValueError(
-                "Unexpected breakdown value '{}'".format(breakdown_opt.value))
+            raise ValueError("Unexpected breakdown value '{}'".format(
+                              breakdown_opt.value))
 
         low, high = map(lambda x: x / 100., range_slider.value)
 
         smallest, largest, num_tasks = ray.global_state._job_length()
         diff = largest - smallest
 
         if time_opt.value == total_time_value:
-            tasks = ray.global_state.task_profiles(
-                start=smallest + diff * low,
-                end=smallest + diff * high)
+            tasks = ray.global_state.task_profiles(start=smallest + diff * low,
+                                                   end=smallest + diff * high)
         elif time_opt.value == total_tasks_value:
             if range_slider.value[0] == 0:
-                tasks = ray.global_state.task_profiles(
-                    num_tasks=int(num_tasks * high),
-                    fwd=True)
+                tasks = ray.global_state.task_profiles(num_tasks=int(
+                                                       num_tasks * high),
+                                                       fwd=True)
             else:
-                tasks = ray.global_state.task_profiles(
-                    num_tasks=int(num_tasks * (high - low)),
-                    fwd=False)
+                tasks = ray.global_state.task_profiles(num_tasks=int(
+                                                       num_tasks *
```

Comment:
indent this by four more spaces, and the next line also (assuming that they can't fit together on the same line)